sdk/python/feast/feature_store.py

```diff
@@ -141,8 +141,10 @@ def refresh_registry(self):
         downloaded synchronously, which may increase latencies if the triggering method is get_online_features()
         """
         registry_config = self.config.get_registry_config()
-        self._registry = Registry(registry_config, repo_path=self.repo_path)
-        self._registry.refresh()
+        registry = Registry(registry_config, repo_path=self.repo_path)
+        registry.refresh()
+
+        self._registry = registry
```

Comment:
What's the reasoning for this refactoring?

Reply:
if there're multiple threads accessing `self._registry` it makes sense to fully construct replacement (actual proto is being loaded in `registry.refresh`) before making it available for everybody else